lucet-runtime/lucet-runtime-internals/src/instance/execution.rs

```diff
@@ -219,26 +312,62 @@ impl KillState {
     }
 }
 
+/// Instance execution domains.
+///
+/// This enum allow us to distinguish how to appropriately terminate an instance.
+///
+/// We can signal in guest code, but must avoid signalling in host code lest we interrupt some
+/// function operating on guest/host shared memory, and invalidate invariants. For example,
+/// interrupting in the middle of a resize operation on a `Vec` could be extremely dangerous.
```

Comment:
I added some documentation for the domain, and the hostcall methods on the kill state, since domain is now much more important w.r.t. terminability. Most of this was borrowed from existing comments etc., but I would also greatly appreciate eyes here :slightly_smiling_face: